projects/plugins/super-cache/inc/preload-notification.js

```diff
@@ -1,18 +1,92 @@
 jQuery( document ).ready( function () {
-	load_preload_status();
-	setInterval( function () {
-		load_preload_status();
-	}, 3000 );
-} );
-
-/**
- *
- */
-function load_preload_status() {
-	jQuery.get( {
-		url: wpsc_preload_ajax.preload_permalink_url + '?' + Math.random(),
-		success: function ( response ) {
-			jQuery( '#preload_status' ).text( response );
-		},
-	} );
-}
+	const { __, sprintf } = wp.i18n;
+
+	// Set the maximum number of entries to show
+	const MAX_PRELOAD_ENTRIES = 5;
+
+	// Set how often to check.
+	const CHECK_INTERVAL = 3000;
+
+	// Get a reference to the log element and the previous log entry
+	const preloadInfoPanel = jQuery( "#wpsc-preload-status" );
+
+	// Abort early if no info panel exists.
+	if ( ! preloadInfoPanel.length ) {
+		return;
+	}
+
+	update_preload_status( wpsc_preload_ajax.preload_status );
+
+	setInterval( () => {
+		jQuery.post(
+			wpsc_preload_ajax.ajax_url,
+			{
+				action: 'wpsc_get_preload_status',
+			},
+			( json ) => {
+				if ( ! json || ! json.success ) {
+					return;
+				}
+
+				update_preload_status( json.data );
+			}
+		);
+	}, CHECK_INTERVAL );
+
+	function update_preload_status( data ) {
+		// Bail early if no data is available.
+		if ( ! data || ( ! data.running && ! data.next && ! data.previous ) ) {
+			return;
+		}
+
+		preloadInfoPanel.empty();
+
+		if ( data.running ) {
+			const panel = jQuery( '<div class="notice notice-warning">' );
+
+			panel.append( jQuery( '<b>' ).text( __( 'Preloading', 'wp-super-cache' ) ) );
+			panel.append( jQuery( '<p>' ).text( __( 'Preloading is currently running.', 'wp-super-cache' ) ) );
+
+			const ul = panel.append( jQuery( '<ul>' ) );
+			for ( const entry of data.history ) {
+				ul.append( jQuery( '<li>' ).text( entry.group + ' ' + entry.progress + ': ' + entry.url ) );
+			}
+
+			preloadInfoPanel.append( panel );
+		} else if ( data.next || data.previous ) {
+			const panel = jQuery( '<div class="notice notice-info">' );
+
+			if ( data.next ) {
+				const diff = data.next - Math.floor( Date.now() / 1000 );
+				const seconds = diff % 60;
+				const minutes = Math.floor( diff / 60 ) % 60;
+				const hours = Math.floor( diff / 3600 ) % 24;
```

Comment:
I got a weird result while waiting for the preload to start
<img width="443" alt="image" src="https://github.com/Automattic/jetpack/assets/11799079/352922f4-ff79-4e7a-a1f1-249f46b9a4b9">

Reply:
I can't replicate this, but I guess it might happen if the next run time is in the past.

So I've added a guard against that in d2dfe131bb, which guarantees the diff between now and the next-run time never dips below 0.